### ISSUE TITLE
Dynamically load dials from most visited sites

### ISSUE THREAD

**Issue by omperus** [fastaddons/GroupSpeedDial#6]
I would like for the dials to load and update themselves from my most visited sites.

When i first installed the addon i got a option to do this manually but now i can't find it anywhere.

Thanks for the great addon.

**Comment by fastaddons** [comment]
Good idea! Let's define how it should work:

So basically you should be able to create (or more like enable) the "Top sites" group that will be synchronized with Firefox "Top sites" list.

Thumbnails should be automatically created every time there is new URL.

The group should behave just like other groups ("Edit group" window should work), except for some features that needs to be disabled:
- no drag and drop
- no cloud sync (every device will have it's own list)
- context menu on dials needs to be reduced to only supported features (no Edit, Clear, Resize, etc...)

Anything else?

This will be harder than I thought, but I will give it a try soon.

**Comment by omperus** [comment]
>So basically you should be able to create (or more like enable) the "Top sites" group that will be synchronized with Firefox "Top sites" list.

>Thumbnails should be automatically created every time there is new URL.

Indeed, the default Firefox "top sites" feature is a good example of the basic functionality.

>Anything else?

Maybe a option to prevent some domains to appear on the top sites group (by right click menu on dial you don't want or by manual input of domain on settings page), this way you can keep prevent sites from appearing on the group without having to delete your actual Firefox history.

>This will be harder than I thought, but I will give it a try soon.

I hope you are successful so i can get rid of these ugly default firefox new tab tiles, thanks again for the fast response and for your great work.

**Comment by fastaddons** [comment]
It seems that there is no API for removing a page, I can only retrieve the list of pages:
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/topSites
(I will create a feature request for this in Bugzilla if it's not there yet)

I could implement simple hiding myself, but the list of pages is limited (12 items). So then you would see only 11. 
The only way would to open "about:home" (manually, addons cannot open it) and using the "..." / "Dismiss" button.

I'm really busy right now with 4 other tasks (all in progress) so this will have to wait, probably a few weeks, depending on how productive I can get working from home :)
